agent/grpc-external/services/dataplane/get_envoy_bootstrap_params.go

```diff
@@ -85,46 +85,33 @@ func (s *Server) GetEnvoyBootstrapParams(ctx context.Context, req *pbdataplane.G
 			return nil, err
 		}
 
-		// Get all proxy configurations for this workload. Currently we're only looking
-		// for proxy configurations in the same tenancy as the workload.
-		// todo (ishustava): we need to support wildcard proxy configurations as well.
+		computedProxyConfig, err := resource.GetDecodedResource[*pbmesh.ComputedProxyConfiguration](
+			ctx,
+			s.ResourceAPIClient,
+			resource.ReplaceType(pbmesh.ComputedProxyConfigurationType, workloadId))
 
-		proxyCfgList, err := s.ResourceAPIClient.List(ctx, &pbresource.ListRequest{
-			Tenancy: workloadRsp.Resource.Id.GetTenancy(),
-			Type:    pbmesh.ProxyConfigurationType,
-		})
 		if err != nil {
-			logger.Error("Error looking up proxyConfiguration", "error", err)
+			logger.Error("Error looking up ComputedProxyConfiguration for this workload", "error", err)
 			return nil, err
 		}
 
-		// Collect and merge proxy configs.
-		// todo (ishustava): sorting and conflict resolution.
-		bootstrapCfg := &pbmesh.BootstrapConfig{}
-		dynamicCfg := &pbmesh.DynamicConfig{}
-		for _, cfgResource := range proxyCfgList.Resources {
-			var proxyCfg pbmesh.ProxyConfiguration
-			err = cfgResource.Data.UnmarshalTo(&proxyCfg)
-			if err != nil {
-				return nil, status.Errorf(codes.Internal, "failed to parse proxy configuration data: %q", cfgResource.Id.Name)
-			}
-			if isWorkloadSelected(req.ProxyId, proxyCfg.Workloads) {
```

Comment:
The `isWorkloadSelected` function can go now.

Reply:
removed!